static/app/utils/featureObserver.ts

```diff
@@ -1,71 +1,109 @@
 import type {Flags} from 'sentry/types/event';
 import type {Organization} from 'sentry/types/organization';
+import type {Project} from 'sentry/types/project';
 
+export const FEATURE_FLAG_BUFFER_SIZE = 100;
 let __SINGLETON: FeatureObserver | null = null;
 
 export default class FeatureObserver {
   /**
    * Return the same instance of FeatureObserver in each part of the app.
    * Multiple instances of FeatureObserver are needed by tests only.
    */
-  public static singleton() {
+  public static singleton({bufferSize}: {bufferSize?: number}) {
     if (!__SINGLETON) {
-      __SINGLETON = new FeatureObserver();
+      __SINGLETON = new FeatureObserver({bufferSize});
     }
     return __SINGLETON;
   }
 
+  private _bufferSize = 0;
   private FEATURE_FLAGS: Flags = {values: []};
 
+  constructor({bufferSize = FEATURE_FLAG_BUFFER_SIZE}: {bufferSize?: number}) {
+    this._bufferSize = bufferSize;
+  }
```

Comment:
because we talked about optional and defaults and stuff so much, i'd slide the default value up into the singleton. That way, in app code we have the default. But in test code, where we're likely to have different smaller values, the buffer size is a required arg.